core/trino-main/src/main/java/io/trino/security/AccessControl.java

```diff
@@ -222,10 +222,15 @@
     void checkCanShowColumns(SecurityContext context, CatalogSchemaTableName table);
 
     /**
-     * Filter the list of columns to those visible to the identity.
+     * Filter the list of columns to those visible to the identity when executing SHOW COLUMNS, DESCRIBE, etc.
      */
     Set<String> filterColumns(SecurityContext context, CatalogSchemaTableName tableName, Set<String> columns);
 
+    /**
+     * Filter the list of columns to those visible to the identity when querying the table.
+     */
+    Set<String> filterTableSchema(SecurityContext context, QualifiedObjectName tableName, Set<String> columns);
```

Comment:
how is this method different from `filterColumns` above?

- Why do we need two methods?
- What's the occasion where `SELECT *` would want to give you different set of columns than `SHOW COLUMNS FROM ...`